skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+from itertools import product
+from scipy.ndimage import generic_filter
+
+from skimage.util import invert, view_as_windows
+from ._rolling_ball_cy import apply_kernel, apply_kernel_nan
+
+
+def rolling_ellipsoid(image, kernel_size=(100, 100), intensity_vertex=(100,),
+                      has_nan=False):
+    """Estimate background intensity using a rolling ellipsoid.
+
+    The rolling ellipsoid algorithm estimates background intensity for a
+    grayscale image in case of uneven exposure. It is a generalization of the
+    frequently used rolling ball algorithm [1]_.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        The gray image to be filtered.
+    kernel_size: two-element tuple, numeric, optional
+        The length of the spatial vertices of the ellipsoid.
+    intensity_vertex : scalar, numeric, optional
+        The length of the intensity vertex of the ellipsoid.
+    has_nan: bool, optional
+        If ``False`` (default) assumes that none of the values in ``image``
+        are ``np.nan``, and uses a faster implementation.
+
+    Notes
+    -----
+
+    - For the pixel that has its background intensity estimated (wlog. at
+      ``(0,0)``) the rolling ellipsoid method places an ellipsoid under it and
+      raises it until it touches the image umbra at ``pos=(y,x)`` the
+      background intensity is then estimated using the image intensity at that
+      position (``image[*pos]``) plus the difference of ``intensity_vertex``
+      and the intensity of the ellipsoid at ``pos``. The intensity of the
+      ellipsoid is computed using the canonical ellipsis equation::
+
+            semi_spatial = kernel_size / 2
+            semi_vertex = intensity_vertex / 2
+            np.sum((pos/semi_spatial)**2) + (intensity/semi_vertex)**2 = 1
```

Comment:
Indeed, the kernel is every point inside the ellipsoid and its surface. However, here we are interested in the value at the surface because that is the point where the ellipsoid touches the image umbra.

Reply:
Makes sense. :+1:  What do you think about replacing "intensity" with "surface" in the sentence before this block? That might be clearer.

Reply:
done.